### ISSUE TITLE
<figure> not allowed as child of <example>?

### ISSUE THREAD

**Issue by sprossi123** [docbook/docbook#238]
Hello,

sorry if I'm in the wrong place, but I don't know whom to ask about my problem at the moment.

The DocBook Reference for version 5.1 ([https://tdg.docbook.org/tdg/5.1/example.html](https://tdg.docbook.org/tdg/5.1/example.html)) is contradictory, I think, on the contents of `<example>`. In the "Synopsis" and the "Children" section, `<figure>` elements are allowed as children of `<example>`. 

But under "Additional Constraints" it explicitly says that this is not allowed: 
"- figure must not occur among the children or descendants of example".

In the docbook.rng used for validation, it also looks to me like placing `<figure>` inside `<example>` is possible. But the Schematron check with docbook.sch always gives an error and marks the document as not valid.

What am I misunderstanding here? I am grateful for any help.

Thanks,
Jörg

**Comment by ndw** [comment]
The DocBook mailing list is probably a better place for questions like this, but here is ok too.

I can see how the content model of `example` is confusing. In the grammar, all of the "formal objects" (figure, table, example, etc.) are allowed to contain all of the DocBook block elements. It's perhaps unfortunate that the formal objects are in the "all blocks" pattern. Perhaps we should have created an "all blocks except formal objects" pattern for this case. 

But if we had, that would only have fixed the problem at one level. Because `para` is in "all blocks" and `para` can contain `figure`, it would still be necessary to have the Schematron assertion that `figure` is not allowed as a descendant of `example`.

So it was just simpler and easier to maintain if we allowed all blocks.

In general, [formal objets](https://tdg.docbook.org/tdg/5.2/ch02.html#s.formal) are not allowed to nest. That exclusion prevents all sorts of otherwise complicated questions about what it means for a figure with a title to be in an example with a title inside a table with a title.

In short: a valid DocBook document must both conform to the grammar and satisfy the Schematron constraints. You cannot put a figure in an example.

**Comment by sprossi123** [comment]
@ndw Thanks for the quick reply and explanation of the underlying thought regarding nesting of formal objects.
Since we don't necessarily need to rely on the `<title>`, we will be able to solve the structure of the problematic parts with `<informalfigure>`. 
And yes, I'll try to check the mailing list first in the future. 

Thanks,
Jörg
